### ISSUE TITLE
[Bug]: Messenger no longer showing badges

### ISSUE THREAD

**Issue by DominusExult** [ferdium/ferdium-recipes#172]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-recipes/blob/develop/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-recipes/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-recipes/issues) for a feature request that matches the one I want to file, without success.

### Ferdium Version

6.1.0

### Recipe Name

Messenger

### Recipe Version

1.7.3

### Last Known working version of the recipe

1.3.3

### Steps to reproduce

- nuked the settings of Ferdium
- started and selected to continue without making an account
- added Messenger Service and logged in
- added another service and selected that, so that new notifications are shown as badges

### Expected Behavior

A new message in Messenger should show a badge.

### Actual Behavior

No badge is shown when I get a new message

### Screenshots

![FerdiumMessenger](https://user-images.githubusercontent.com/3313896/188489218-f9274fa6-7ff7-4055-b72e-00f947a3a3c9.png)
left: Ferdium 6.1.0
right: Ferdi 5.8.1

### Additional Information

I'm on macOS 12.5. To give a reliable feedback, I deleted both settings folder for Ferdium and Ferdi in ~/Library/Application Support

I also tried Ferdium 6.0 and I don't get the badges on this either. I switched last week from Ferdi to Ferdium and noticed the Messenger badges not working, hence my surprise that it worked in Ferdi but not anymore in Ferdium.

Also note: I did tick the checkboxes for having read contribution guidelines and code of conduct but the links do not go to an existing file.

**Comment by jimmylad** [comment]
I see this too. 
Ferdium 6.1.0
MacOS 12.5

**Comment by jimmylad** [comment]
I have resorted to using code from https://github.com/getferdi/recipes/blob/24cccb0a6be445c155ad2947789898f5d19b1789/recipes/messenger/webview.js 
lines 2-16

**Comment by DominusExult** [comment]
I can confirm that:

> I have resorted to using code from https://github.com/getferdi/recipes/blob/24cccb0a6be445c155ad2947789898f5d19b1789/recipes/messenger/webview.js
> lines 2-16

**Comment by HugLifeTiZ** [comment]
I had this working with some local changes, but the update with commit 39c3f63261a5befd9f2ed42b27bd5ebeb0b8559c broke it for me, in the sense that it shows a badge for muted chats.

Taking the getMessages function from the above link brings the functionality back to as desired.

**Comment by Alphrag** [comment]
This change doesn't seem to work for me (I have just started looking into this with the change of ui). Could you all try to replace the function `getMessages` of the `webview.js` for your messenger recipe by the following one instead and tell me if this does the job?

```js
const getMessages = () => {
   
    let count = 0;
    /*
     * add count of unread messages from messenger inner counter
     */
    const unreadCounter = document.querySelector('a[aria-label$="unread"]');
    if (unreadCounter) {
      count += Ferdium.safeParseInt(unreadCounter.ariaLabel.split(' ')[1]);
    }

    /*
     * add count of message requests on top of unread counter
     */
    const messageRequestsElement = document.querySelector('._5nxf');
    if (messageRequestsElement) {
      count += Ferdium.safeParseInt(messageRequestsElement.textContent);
    }
   
    Ferdium.setBadge(count);
  };
```
The only thing I am not completely sure is about the message requests one as it is still using the css name of before and this might have changed with the redesign.

**Comment by DominusExult** [comment]
Yes, your change works for me. The first workaround stopped working for me a few days before your post.

**Comment by Alphrag** [comment]
Ok good to know. I didn't see that there was already a PR using the same type of parsing. I will modify it to make both workarounds available in case.

**Comment by Alphrag** [comment]
@jimmylad can you tell me if you still have the old UI, and if yes, in which country are you/which language are you using? I have tried many places over the world (with a VPN) and many different languages and I have always had the new UI. So if you still have the old one, that would be handy to test things (if you can extract the html of the page, in particular)

**Comment by jimmylad** [comment]
@Alphrag same as @DominusExult my change stopped working and yours is now working. Thanks

**Comment by Alphrag** [comment]
This should now be resolved for everyone following #145.
